pkg/azuredx/azureauth/scopes.go

```diff
@@ -0,0 +1,35 @@
+package azureauth
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/grafana/grafana-azure-sdk-go/azcredentials"
+	"github.com/grafana/grafana-azure-sdk-go/azsettings"
+)
+
+var (
+	azureDataExplorerScopes = map[string][]string{
+		azsettings.AzurePublic:       {"https://kusto.kusto.windows.net/.default"},
+		azsettings.AzureChina:        {"https://kusto.kusto.chinacloudapi.cn/.default"},
+		azsettings.AzureUSGovernment: {"{clusterUrl}/.default"},
+	}
+)
+
+func getAzureScopes(credentials *azcredentials.AzureClientSecretCredentials, clusterUrl string) ([]string, error) {
```

Comment:
In this function, could you have addressed TODO #356 by checking clusterUrl for special case top domains for `AzurePublic` and `AzureChina` and if neither of those, then simply applying the generalized substitution 
`[]string{fmt.Sprintf("%s/.default", clusterUrl)}`?

Or does that not protect against the free input of cluster URL as you mentioned?

Reply:
This needs additional testing, I just don't want to address it in this PR to not cause unwanted regressions on top of the significant changes I made. I preserved the original logic and created task to address it separately.

Reply:
Got it.

>I preserved the original logic

It looks like the original `getCloudScope` did not return [this error case for unknown Azure cloud on line 24](https://github.com/grafana/azure-data-explorer-datasource/pull/358/files#diff-98f65fb6bc9c4660e45ec0766e5bf93e54d5cd21eb6443db20f18105a5c5c173R24) unless I'm missing something.

Reply:
I meant I preserved the scope values as they were before: Public/China - predefined string, US Gov - cluster URL.
I added error because falling back to public cloud value doesn't make sense, it won't work, and inconsistent with other parts which have logic based on selected cloud ([here](https://github.com/grafana/grafana-azure-sdk-go/blob/e079ca81c3fb6b309b37843345cf97f30aad4609/aztokenprovider/token_provider.go#L124) and  [here](https://github.com/kostrse/azure-data-explorer-datasource/blob/2076be1224c340329c2a3a3fa1cdc70b6d78fd0d/pkg/azuredx/azureauth/auth.go#L214)).